tsl/src/bgw_policy/retention_api.c

```diff
@@ -230,28 +230,23 @@ policy_retention_add(PG_FUNCTION_ARGS)
 	{
 		case INTERVALOID:
 			ts_jsonb_add_interval(parse_state,
-								  CONFIG_KEY_RETENTION_WINDOW,
+								  CONFIG_KEY_DROP_AFTER,
 								  DatumGetIntervalP(window_datum));
 			break;
 		case INT2OID:
-			ts_jsonb_add_int64(parse_state,
-							   CONFIG_KEY_RETENTION_WINDOW,
-							   DatumGetInt16(window_datum));
+			ts_jsonb_add_int64(parse_state, CONFIG_KEY_DROP_AFTER, DatumGetInt16(window_datum));
 			break;
 		case INT4OID:
-			ts_jsonb_add_int64(parse_state,
-							   CONFIG_KEY_RETENTION_WINDOW,
-							   DatumGetInt32(window_datum));
+			ts_jsonb_add_int64(parse_state, CONFIG_KEY_DROP_AFTER, DatumGetInt32(window_datum));
 			break;
 		case INT8OID:
-			ts_jsonb_add_int64(parse_state,
-							   CONFIG_KEY_RETENTION_WINDOW,
-							   DatumGetInt64(window_datum));
+			ts_jsonb_add_int64(parse_state, CONFIG_KEY_DROP_AFTER, DatumGetInt64(window_datum));
 			break;
 		default:
 			ereport(ERROR,
 					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-					 errmsg("unsupported datatype for retention_window: %s",
+					 errmsg("unsupported datatype for %s: %s",
```

Comment:
nit: data type

Reply:
There are several error messages that need an overhaul, including those that don't conform with the error style guide. I propose leaving this to a separate PR.